fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal

```diff
@@ -31,63 +41,92 @@ public isolated class Uscore501PatientSourceConnect {
+                return fhirError;
+            } else {
+                uscore501:USCorePatientProfile|error fhirResource = payload.cloneWithType(uscore501:USCorePatientProfile);
+                
+                if (fhirResource is error) {
+                    r4:FHIRError fhirError = r4:createFHIRError("Did not get a FHIR Resource from source.", r4:CODE_SEVERITY_ERROR,r4:TRANSIENT_EXCEPTION);
+                    return fhirError;
+                } else {
+                    return fhirResource;
+                }
+            }
+        }
     }
 
     isolated function search(map<r4:RequestSearchParameter[]> searchParameters, r4:FHIRContext ctx) returns r4:Bundle|Patient[]|r4:FHIRError {
 
-        uscore501:USCorePatientProfile[] patients = [];
-
-        //Implement source system connection here and retreive data.
-        //Create FHIR resource from retreived data.
-        uscore501:USCorePatientProfile example = {
-            id: "12d39",
-            meta: {
-                versionId: "abc12s3",
-                profile: ["http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"]
-            },
-            identifier: [
-            ],
-            implicitRules: "https://www.hl7.org/fhir",
-            language: "en-US"
-        ,gender: "unknown", name: []};
-        uscore501:USCorePatientProfile example1 = {
-            id: "12c39",
-            meta: {
-                versionId: "abc12s3",
-                profile: ["http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"]
-            },
-            identifier: [
-            ],
-            implicitRules: "https://www.hl7.org/fhir",
-            language: "en-US"
-        ,gender: "unknown", name: []};
-        patients.push(example);
-        patients.push(example1);
-        return patients;
+        //convert search parameters to map<string|string[]>
+        map<string|string[]> searchParams = {};
+        foreach var [key, value] in searchParameters.entries() {
+            foreach var param in value {
+                searchParams[key] = param.value;
+            }
+        }
+        //convert search parameters to query string
+        string queryString = "";
+        foreach var [key, value] in searchParams.entries() {
+            // check if value is an array
+            if (value is string[]) {
+                foreach var v in value {
+                    queryString = queryString + key + "=" + v + "&";
+                }
+            } else {
+                queryString = queryString + key + "=" + <string>value + "&";
+            }
+        }
+        //remove last & if query string is not empty
+        if (queryString != "") {
+            queryString = "?" + queryString.substring(0, queryString.length() - 1);
+        }
+
+        http:Response|http:ClientError res = sourceEp->get(SEARCH + queryString);
```

Comment:
Ballerina encourage to use resource access action if possible. https://ballerina.io/learn/by-example/http-client-send-request-receive-response/

i.e: by default it is `get` and hence verb get is not required.